### PR TITLE
(PC-1536) extract method and allfilter by remaining for searched reco…

### PR DESCRIPTION
**Pull request by annemarie35**
…mmendations

### CODE REVIEW

**Comment by sofcalca** [comment]
il faudra rebase la branche

File: tests/routes/search_test.py [hunk 1]
```diff
@@ -725,3 +725,32 @@ def when_searching_by_matching_date_and_non_matching_keywords(self, app):
             # Then
             assert response.status_code == 200
             assert len(response.json()) == 0
+
+        @clean_database
+        def when_searching_by_matching_keywords_but_recommendations_has_no_stock(self, app):
```
Comment (abel-andre) [inline tests/routes/search_test.py]: J'ai l'impression que c'est un test qui devrait se faire sur la query et non sur la route parce que c'est un cas aux limites (_edge case_) et qu'il est trop spécifique pour un test d'API. Qu'en pensez-vous ?

Comment (sofcalca) [reply tests/routes/search_test.py]: oui en effet
